test/t8_forest_incomplete/t8_test_elements_remove.cxx

```diff
@@ -0,0 +1,289 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
```

Comment:
Please document this file, all functions and update all variables to our coding guidelines. 
This Test should be a parametrized google-test.

So far all your tests are rather complex, which is good, but we should also add a very simple basic test, which will make it easier to debug in the future. I suggest to construct a single element, refine it once and then delete between 0 and `num_children-1` elements. That way we can:
- check if the elements are deleted
- also check `is_incomplete_family`

Reply:
Done. I have removed this complex test. See `t8_gtest_permute_hole.cxx` instead.